### PR TITLE
[Merged by Bors] - feat: port Topology.Category.Compactum

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
What's left:

- [ ] computability issues (see porting notes)
- [ ] Failing to synthesize instance  TopologicalSpace X.A for Compactum X (and a few consequences of that)
- [ ] Manual naming, i.e. `continuous_of_hom` should probably be `continuousOfHom` and names in comments

**Comment by mo271** [comment]
https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/!4.233872.20Topology.2ECategory.2ECompactum

File: Mathlib/Topology/Category/Compactum.lean [hunk 1]
```diff
@@ -74,7 +74,8 @@ We also add wrappers around structures which already exist. Here are the main on
 
 -/
 
-
+-- porting note: "Compactum" is already uppser case
```
Comment (jcommelin) [inline Mathlib/Topology/Category/Compactum.lean:77]:
```suggestion
-- porting note: "Compactum" is already upper case
```

**Pull request closed without merging**